// @sbaiahmed1/react-native-blur/lib/module/LiquidGlassView.js
"use strict";

import React from 'react';
import { Platform } from 'react-native';
import ReactNativeLiquidGlassView from './ReactNativeLiquidGlassViewNativeComponent';
import BlurView from "./BlurView.js";
import { jsx as _jsx } from "react/jsx-runtime";
/**
 * A Liquid Glass view component that provides iOS 26+ glass effects.
 *
 * This component uses the new UIKit glass effect API available on iOS 26+.
 * On older iOS versions or when reduced transparency is enabled, it falls back
 * to a solid color background.
 *
 * **Platform Support:**
 * - iOS 26+: Native glass effect with full functionality
 * - iOS < 26 or Android: Fallback to reducedTransparencyFallbackColor
 *
 * This component automatically handles the proper positioning pattern where the glass
 * effect is positioned absolutely behind the content, ensuring interactive elements
 * work correctly.
 *
 * @example
 * ```tsx
 * import { LiquidGlassView } from '@sbaiahmed1/react-native-blur';
 *
 * <LiquidGlassView
 *   glassType="clear"
 *   glassTintColor="#007AFF"
 *   glassOpacity={0.8}
 *   style={{ flex: 1 }}
 * >
 *   <Text>Content on top of liquid glass</Text>
 *   <Button title="Interactive Button" onPress={() => {}} />
 * </LiquidGlassView>
 * ```
 *
 * @platform ios
 */
export const LiquidGlassView = ({
  glassType = 'clear',
  glassTintColor = 'clear',
  glassOpacity = 1.0,
  reducedTransparencyFallbackColor = '#FFFFFF',
  isInteractive = true,
  ignoreSafeArea = true,
  style,
  children,
  ...props
}) => {
  const isIos = Platform.OS === 'ios';

  // Only render on iOS 26+ (fallback otherwise)
  if (!isIos || isIos && Number.parseInt(String(Platform.Version), 10) < 26) {
    console.warn('LiquidGlassView is only supported on iOS. Rendering children without glass effect.');

    // Compute overlay color with opacity for Android native handling
    const overlayColorWithAlpha = glassTintColor + Math.floor(glassOpacity * 255).toString(16).padStart(2, '0');
    return /*#__PURE__*/_jsx(BlurView, {
      blurAmount: 70,
      overlayColor: overlayColorWithAlpha,
      reducedTransparencyFallbackColor: reducedTransparencyFallbackColor,
      ignoreSafeArea: ignoreSafeArea,
      style: style,
      children: children
    });
  }

  // If children exist, use the absolute positioning pattern
  return /*#__PURE__*/_jsx(ReactNativeLiquidGlassView, {
    glassType: glassType,
    glassTintColor: glassTintColor,
    glassOpacity: glassOpacity,
    reducedTransparencyFallbackColor: reducedTransparencyFallbackColor,
    isInteractive: isInteractive,
    ignoreSafeArea: ignoreSafeArea,
    style: style,
    ...props,
    children: children
  });
};
export default LiquidGlassView;
