// @sbaiahmed1/react-native-blur/lib/module/VibrancyView.js
"use strict";

import React from 'react';
import { Platform, StyleSheet } from 'react-native';
import ReactNativeVibrancyView from './ReactNativeVibrancyViewNativeComponent';
import BlurView from "./BlurView.js";
import { jsx as _jsx } from "react/jsx-runtime";
/**
 * A component that applies a vibrancy effect to its content.
 *
 * On iOS, this uses UIVibrancyEffect.
 * On Android, this falls back to a simple View (or BlurView behavior if implemented).
 */
export const VibrancyView = ({
  blurType = 'xlight',
  blurAmount = 10,
  style,
  children,
  ...props
}) => {
  if (Platform.OS === 'android') {
    return /*#__PURE__*/_jsx(BlurView, {
      blurType: blurType,
      blurAmount: blurAmount,
      style: style,
      ...props,
      children: children
    });
  }
  return /*#__PURE__*/_jsx(ReactNativeVibrancyView, {
    blurType: blurType,
    blurAmount: blurAmount,
    style: [styles.container, style],
    ...props,
    children: children
  });
};
const styles = StyleSheet.create({
  container: {
    backgroundColor: 'transparent'
  }
});
